src/main/java/com/squareup/javapoet/JavaFile.java

```diff
@@ -40,6 +42,22 @@
 
 /** A Java file containing a single top level class. */
 public final class JavaFile {
+  static final class StaticImportItem implements Comparable<StaticImportItem> {
+    final ClassName className;
+    final String memberName;
+    final String signature;
```

Comment:
Needed the canonicle class name for indentification of potentially imported `$T`s ... will try to boil it down to the signature string, as the `if (staticImportClassNames.contains(typeName))` condition in `case "$T":` could be written as `if (staticImportSignatures.contains(typeName.toString()))`